includes/in-person-payments/class-wc-payments-in-person-payments-receipts-service.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Class WC_Payments_In_Person_Payment_Print_Receipt_Service
+ *
+ * @package WooCommerce\Payments
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class handling in person payments receipts.
+ */
+class WC_Payments_In_Person_Payments_Receipts_Service {
+
+	/**
+	 * Renders the receipt template.
+	 *
+	 * @param  array    $settings Merchant settings.
+	 * @param  WC_Order $order Order instance.
+	 * @param  array    $charge Charge data.
+	 *
+	 * @return string
+	 */
+	public function get_receipt_markup( array $settings, WC_Order $order, array $charge ) :string {
+		$this->validate_settings( $settings );
+		$this->validate_charge( $charge );
+
+		$order_data = [
+			'id'           => $order->get_id(),
+			'currency'     => $order->get_currency(),
+			'subtotal'     => $order->get_subtotal(),
+			'line_items'   => $order->get_items(),
+			'coupon_lines' => $order->get_items( 'coupon' ),
+			'tax_lines'    => $order->get_items( 'tax' ),
+			'total'        => $order->get_total(),
+		];
+
+		ob_start();
+
+		wc_get_template(
+			'html-in-person-payment-receipt.php',
+			[
+				'amount_captured'        => $charge['amount_captured'] / 100,
+				'coupon_lines'           => $order_data['coupon_lines'] ?? [],
+				'business_name'          => $settings['business_name'],
+				'line_items'             => $this->format_line_items( $order_data ),
+				'order'                  => $order_data,
+				'payment_method_details' => $charge['payment_method_details']['card_present'],
+				'receipt'                => $charge['payment_method_details']['card_present']['receipt'],
+				'support_address'        => $settings['support_info']['address'],
+				'support_email'          => $settings['support_info']['email'],
+				'support_phone'          => $settings['support_info']['phone'],
+				'tax_lines'              => $order_data['tax_lines'] ?? [],
+			],
+			'',
+			__DIR__ . '/templates/'
+		);
+
+		return ob_get_clean();
+	}
+
+	/**
+	 * Format line items
+	 *
+	 * @param  array $order the order.
+	 * @return array
+	 */
+	private function format_line_items( array $order ) :array {
+		$line_items_data = [];
+
+		foreach ( $order['line_items'] as $item ) {
+			$item_data            = $item->get_data();
+			$item_data['product'] = $item->get_product()->get_data();
+			$line_items_data[]    = $item_data;
+		};
+
+		return $line_items_data;
+	}
+
+	/**
+	 * Validate settings
+	 *
+	 * @param  array $settings Settings.
+	 * @return void
+	 * @throws \Exception Error validating settings.
+	 */
+	private function validate_settings( array $settings ) {
+		if ( ! $settings ) {
+			throw new \Exception( 'You must provide settings information.' );
```

Comment:
`You must provide` might be `shouting` for some people. `%s needs to be provided.` would be nicer.